tls/s2n_record_read.c

```diff
@@ -106,12 +106,20 @@ int s2n_record_header_parse(
     S2N_ERROR_IF(conn->actual_protocol_version_established && MIN(conn->actual_protocol_version, S2N_TLS12) /* check against legacy record version (1.2) in tls 1.3 */
                             != version,
             S2N_ERR_BAD_MESSAGE);
-    POSIX_GUARD(s2n_stuffer_read_uint16(in, fragment_length));
 
-    /* Some servers send fragments that are above the maximum length.  (e.g.
-     * Openssl 1.0.1, so we don't check if the fragment length is >
-     * S2N_TLS_MAXIMUM_FRAGMENT_LENGTH. The on-the-wire max is 65k
+    /* Some servers send fragments that are above the maximum length (e.g.
+     * Openssl 1.0.1), so we don't check if the fragment length is >
+     * S2N_TLS_MAXIMUM_FRAGMENT_LENGTH. The on-the-wire max is ~65k so
+     * process up to that limit instead.
```

Comment:
https://github.com/aws/s2n-tls/pull/4202#discussion_r1329385745 Sounds like we should just use the limit we're actually enforcing here. It has to fit in a uint16, so 2^16. Unless we perform additional validation of this length later?
```suggestion
     * S2N_TLS_MAXIMUM_FRAGMENT_LENGTH. We allow up to 2^16.
```

Reply:
Looking at the code and from what I can tell we dont perform other validation on the length, which aligns with the existing comment here.

I like your version where we state current behavior. I was trying to preserve the phrase "on-the-wire" but that seems confusing.